lib/spack/spack/compilers/msvc.py

```diff
@@ -264,6 +286,22 @@ def vs_root(self):
         # this should be consistent accross versions
         return os.path.abspath(os.path.join(self.cc, "../../../../../../../.."))
 
+    def run_vc_vars(self, sdk_ver=None):
+        # provide vcvars with msvc version selected by concretization,
+        # not whatever it happens to pick up on the system (highest available version)
+
+        # get current platform architecture and format for vcvars argument
+        arch = spack.platforms.real_host().default.lower()
+        arch = arch.replace("-", "_")
+        out = subprocess.check_output(  # novermin
+            'cmd /u /c "{}" {} {} {} && set'.format(
+                self.setvarsfile, arch, sdk_ver, "-vcvars_ver=%s" % self.msvc_version
+            ),
+            stderr=subprocess.STDOUT,
```

Comment:
It's what we're doing in develop right now, I just copied it. I think this is moot since I'll be taking this out once we make the compiler wrapper an independent package.